dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "PushNotificationManager.h"
+#include "Microsoft.Windows.PushNotifications.PushNotificationManager.g.cpp"
+
+#include "PushNotificationCreateChannelResult.h"
+
+#include <winrt/Windows.ApplicationModel.background.h>
+#include <winrt/Windows.Networking.PushNotifications.h>
+#include "PushNotificationBackgroundTask.h"
+
+#include <winerror.h>
+#include <algorithm>
+#include "PushNotificationChannel.h"
+#include "externs.h"
+#include <string_view>
+
+using namespace std::literals;
+
+constexpr std::wstring_view backgroundTaskName = L"PushBackgroundTaskName"sv;
+
+namespace winrt
+{
+    using namespace Windows::ApplicationModel::Background;
+    using namespace Windows::Networking::PushNotifications;
+    using namespace Windows::Foundation;
+}
+
+namespace winrt::Microsoft::Windows::PushNotifications::implementation
+{
+    inline constexpr auto c_maxBackoff{ 5min };
+    inline constexpr auto c_initialBackoff{ 60s };
+    inline constexpr auto c_backoffIncrement{ 60s };
+
+    const HRESULT WNP_E_NOT_CONNECTED = (HRESULT)0x880403E8L;
```

Comment:
constexpr HRESULT WNP_E_NOT_CONNECTED{ 0x880403E8L }; - conversion from 'unsigned long' to 'const HRESULT' requires a narrowing conversion. So I went with the static_cast approach here. Thanks.

Reply:
> constexpr HRESULT WNP_E_NOT_CONNECTED{ 0x880403E8L }; - conversion from 'unsigned long' to 'const HRESULT' requires a narrowing conversion. So I went with the static_cast approach here. Thanks.

Probably because long>int. But HRESULT's a typedef for LONG so I bet `0x880403E8` would be fine.

For future reference

Reply:
It doesn't work because `0x880403E8L` does not fit in a `LONG`. The largest `LONG` is `0x7FFFFFFFL`. Therefore, it overflows to an `unsigned long`. Writing `0x880403E8` would also be an `unsigned long` for the same reason (too big to fit in a `long`.) You have to use a `static_cast` to tell the compiler "Yes, I intentionally want to trigger integer overflow."